Zend/zend_inheritance.c

```diff
@@ -606,31 +606,28 @@ static inheritance_status zend_perform_covariant_type_check(
 	bool have_unresolved = false;
 
 	if (ZEND_TYPE_IS_INTERSECTION(fe_type)) {
-		/* As an intersection type can only be composed of class types (checked at compile stage),
-		 * it is trivially covariant to the object type.
-		 * Handle this case separately to ensure it never requires class loading. */
-		if (proto_type_mask & MAY_BE_OBJECT) {
-			return INHERITANCE_SUCCESS;
-		}
-
-		// TODO: Make "iterable" an alias of "array|Traversable" instead,
-		// so this special cases will be handled automatically.
-		if (proto_type_mask & MAY_BE_ITERABLE) {
+		/* Currently, for object type any class name would be allowed here.
+		 * We still perform a class lookup for forward-compatibility reasons,
+		 * as we may have named types in the future that are not classes
+		 * (such as enums or typedefs). */
```

Comment:
The "enums" bit here doesn't make sense anymore :)

Reply:
Copy pasted this, so I also fixed that comment at the same time.